runtime/parachains/src/inclusion/mod.rs

```diff
@@ -792,6 +918,85 @@ impl<T: Config> Pallet<T> {
 			)
 	}
 
+	/// Check that all the upward messages sent by a candidate pass the acceptance criteria. Returns
+	/// false, if any of the messages doesn't pass.
+	// FAIL-CI: TODO update docs
+	pub(crate) fn check_upward_messages(
+		config: &HostConfiguration<T::BlockNumber>,
+		para: ParaId,
+		upward_messages: &[UpwardMessage],
+	) -> Result<(), UmpAcceptanceCheckErr> {
+		// Cannot send UMP messages while off-boarding.
+		if <paras::Pallet<T>>::lifecycle(para).map_or(false, |l| l.is_offboarding()) {
+			ensure!(upward_messages.is_empty(), UmpAcceptanceCheckErr::IsOffboarding,);
```

Comment:
Here it just rejects candidates with UMP messages while the para is off-boarding.  
Is this good or should we let it pass and silently drop in `enact_candidate`?

Reply:
I don't think this is new logic, and it doesn't seem unreasonable.